### PR TITLE
feat: copy db dump to s3

### PR DESCRIPTION
**Pull request by MDunitz**
- #TICKET_NUMBER

### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add lambda code to copy a dump of the rds db to  a s3 bucket

### CODE REVIEW

File: .happy/terraform/modules/ecs-stack/main.tf [hunk 1]
```diff
@@ -237,6 +238,26 @@ module dataset_submissions_lambda {
   security_groups            = local.security_groups
 }
 
+
+module "prod_data_copy" {
+ # this ensures this resource is only created in prod
+ # todo uncomment out next line when confirmed to work in dev
+ # count                 = local.deployment_stage == "prod" ? 1 : 0
```
Comment (ebezzi) [inline .happy/terraform/modules/ecs-stack/main.tf:245]:
```suggestion
```

File: backend/corpora/sync_env_with_prod/app.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import subprocess
+
+
+def copy_relational_db(event, context):
+    print('Starting copy')
+    db_uri = generate_db_uri()
+    copy_rds_data(db_uri)
+    # todo update to write to prod after confirming code works in dev
+    write_to_s3('cellxgene-db-dump-dev')
+    return {
```
Comment (ebezzi) [inline backend/corpora/sync_env_with_prod/app.py]: Maybe add a try/catch and return an appropriate error? Not super necessary but could help with debugging issues

File: backend/corpora/sync_env_with_prod/app.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import json
```
Comment (atolopko-czi) [inline backend/corpora/sync_env_with_prod/app.py:1]: maybe rename file to `dump_db.py` now that we have `load_db.py` sibling module?

**Review by atolopko-czi — COMMENTED** [review]
Just some minor comments; not a proper review.

File: backend/Makefile [hunk 4]
```diff
@@ -97,3 +97,10 @@ db/tunnel:
 	ssh -f -T -N \
 		-o ServerAliveInterval=${SSH_SERVER_ALIVE_INTERVAL_IN_SECONDS} -o ServerAliveCountMax=${SSH_SERVER_ALIVE_COUNT_MAX} \
 		-L 5432:${endpoint}:5432 bastion.${DEPLOYMENT_STAGE}.single-cell.czi.technology
+
+db/download-prod:
```
Comment (atolopko-czi) [inline backend/Makefile:101]: still need this?

File: Dockerfile.sync_envs_with_prod [hunk 5]
```diff
@@ -0,0 +1,17 @@
+ARG BASE_TAG=branch-main
```
Comment (atolopko-czi) [inline Dockerfile.sync_envs_with_prod:1]: re: file name, we're using both "mirror" and "sync" terms and we should probably settle on one for consistency amongst all related code.  For the load lambda, I was going to call it `mirror-prod-env-data-to-<env>`, e.g.  I don't feel strongly about any particular name, just the consistency. :)

Comment (atolopko-czi) [reply Dockerfile.sync_envs_with_prod:1]: (we can circle back on naming at the end, once we have a working system.)

File: backend/Makefile (hunk 4 above)
Comment (MDunitz) [reply backend/Makefile:101]: I was considering leaving it in as a convenience method to grab the prod db locally. What do you think?

**Comment by atolopko-czi** [comment]
Abandoned effort for lambda-based mirroring. Closing.

**Pull request closed without merging**